core/kazoo_apps/src/kapps_notify_publisher.erl

```diff
@@ -167,20 +167,25 @@ maybe_ignore_failure(_) -> 'false'.
 
 %% @private
 %% @doc try to find account id in different part of payload(copied from teletype_util)
--spec find_account_id(api_terms()) -> api_binary().
+-spec find_account_id(api_terms()) -> api_ne_binary().
+find_account_id(Req) when is_list(Req) ->
+    find_account_id(Req, fun props:get_first_defined/2);
 find_account_id(Req) ->
-    props:get_first_defined([<<"account_id">>
-                            ,[<<"account">>, <<"_id">>]
-                            ,<<"pvt_account_id">>
-                            ,<<"_id">>, <<"id">>
-                            ,<<"Account-ID">>
-                            ,[<<"details">>, <<"account_id">>]
-                            ,[<<"Details">>, <<"Account-ID">>]
-                            ,[<<"details">>, <<"custom_channel_vars">>, <<"account_id">>]
-                            ,[<<"Details">>, <<"Custom-Channel-Vars">>, <<"Account-ID">>]
-                            ]
-                           ,Req
-                           ).
+    find_account_id(Req, fun kz_json:get_first_defined/2).
+
+find_account_id(Req, Get) ->
+    Get([<<"account_id">>
+        ,[<<"account">>, <<"_id">>]
+        ,<<"pvt_account_id">>
+        ,<<"_id">>, <<"id">>
+        ,<<"Account-ID">>
+        ,[<<"details">>, <<"account_id">>]
+        ,[<<"Details">>, <<"Account-ID">>]
+        ,[<<"details">>, <<"custom_channel_vars">>, <<"account_id">>]
+        ,[<<"Details">>, <<"Custom-Channel-Vars">>, <<"Account-ID">>]
+        ]
```

Comment:
Yes but with your changes `find_account_id/2` has to do more than it needs to at runtime.
I am suggesting hand-unrolling the 2 possible cases explicitly so that the code generated by the compiler does not need to call `erlang:apply/3` at all.
I'm saying let's put this list of paths in a `/0` function and write

```erlang
find_account_id(Req) when is_list(Req) ->
    props:get_first_defined(account_id_paths(), Req);
find_account_id(Req) ->
    kz_json:get_first_defined(account_id_paths(), Req).
```